temporal-sdk/src/main/java/io/temporal/worker/WorkerOptions.java

```diff
@@ -90,9 +91,9 @@ private Builder(WorkerOptions o) {
         return;
       }
       this.maxWorkerActivitiesPerSecond = o.maxWorkerActivitiesPerSecond;
-      this.maxConcurrentActivityExecutionSize = o.maxConcurrentActivityExecutionSize;
-      this.maxConcurrentWorkflowTaskExecutionSize = o.maxConcurrentWorkflowTaskExecutionSize;
-      this.maxConcurrentLocalActivityExecutionSize = o.maxConcurrentLocalActivityExecutionSize;
+      this.workflowSlotSupplier = o.workflowSlotSupplier;
+      this.activitySlotSupplier = o.activitySlotSupplier;
+      this.localActivitySlotSupplier = o.localActivitySlotSupplier;
```

Comment:
I agree with 1. , I  don't have a strong opinion on 2 , but I slightly prefer 3 separate options for each `SlotSupplier` since it fits with a builder pattern better and handle a mix of `SlotSupplier`  and legacy config.

Reply:
Think about it from a user POV. What is the simplest configuration for a user to choose "resource limited tuning"? I am happy we didn't make a user set 3 different interceptor fields for use of OTel interceptors. It's not that the 3 things can still be provided, it's just from a user and default impl, people shouldn't have to set three things if they don't need to.

Reply:
I think it's dramatically easier for the user to provide separate suppliers for separate kinds, which is I think more likely the more common case than a shared implementation across different kinds - and it makes that case no harder. Setting two more options fields is effectively zero effort. 

If it was one option, we either have to have something like `SlotSupplier.activity(a).workflow(b).localActivity(c)` which is now the same thing, but with one _extra_ step of putting that whole thing in the worker config. _Or_ we have to change the whole interface to pass the type on every reservation call which I think is much messier and conceptually more difficult for most implementations.

I'll change the options to keep the old values, though.

Reply:
Plus, since we now already have mutually exclusive options, there's zero reason we can do `builder().setResourceLimitedTuner(options)` and under the hood it sets all there.

Reply:
I'm not totally sure 1. makes sense to me either, honestly, as it requires establishing precedence rules for these rather than last-call-wins, but you both seem to agree so I'll make the change.

Reply:
> If it was one option, we either have to have something like `WorkerTuner.activity(a).workflow(b).localActivity(c)`

It'd be more like `builder().setWorkerTuner(new TheTuner(a, b, c))` which is nice because people can extend that one or get one from a third part lib or whatever

> Plus, since we now already have mutually exclusive options, there's zero reason we can't do `builder().setResourceLimitedTuner(options)` and under the hood it sets all three.

That means only we get the single-option approach because we are Temporal. Everyone should be allowed to have users use _their_ tuner via a single option too. In the meantime, they may have to provide a static `applyToWorkerOptionsBuilder` to make sure all three suppliers are set right.

Reply:
I don't think that's an apples-to-apples comparison at all though. Your tracing interceptor isn't making decisions about what happens where the first and most important criteria is the type of slot - rather it is dealing with application-wide traces, and may or may not care about the specifics of the thing being traced. Here, you _always_ care about the type of slot.

Reply:
The implementation always cares about the type of thing it's intercepting or the slot, the one setting the worker option does not. Well, they may in some cases (e.g. fixed size things they may provide three options to this tuner), but they may not in other cases (e.g. they only want to provide resource limits to tuner or just leave defaults). But it should be up to the tuner implementation to decide whether the configurer cares about separate things or not (the implementation itself of course will always care about separate things).

Even in cases where you may want to mix (e.g. fixed and resource-limited), I think still requiring an impl that returns different slot suppliers for those specific cases is better UX than making _everyone_ care about those separate slot suppliers in _every_ case.

I think it may be worth seeing how a user would provide their own implementation of a resource tuner that needs to work across the three (we shouldn't be special in that we're the only ones that get single-option config).